File: be/src/modules/auth/dto/login.dto.ts

import { ApiProperty, ApiPropertyOptional } from '@nestjs/swagger';
import { IsEmail, IsNotEmpty } from 'class-validator';

export class LoginDTO {
  @IsEmail({}, { message: 'Email is invalid' })
  @ApiProperty()
  email: string;

  @IsNotEmpty({ message: 'Password is required' })
  @ApiProperty()
  password: string;

  loginType: number;

  @ApiPropertyOptional()
  clientId?: string;
}

File: hitalive/lib/dto/login.dto.dart
class LoginDTO {
  final String email;
  final String password;

  LoginDTO({required this.email, required this.password});

  Map<String, String> toJson() {
    final Map<String, String> data = <String, String>{};
    data['username'] = email;
    data['password'] = password;
    return data;
  }

  static LoginDTO fromJson(Map<String, dynamic> json) {
    return LoginDTO(
      email: json['email'] ?? '',
      password: json['password'] ?? '',
    );
  }
}
